validation.go

```diff
@@ -13,12 +13,18 @@ const MaxUint = ^uint(0)
 // to expectedRoot.
 func ValidatePartialTree(leafIndices []uint64, leaves, proof [][]byte, expectedRoot []byte,
 	hash HashFunc) (bool, error) {
+	valid, _, err := ValidatePartialTreeWithParkingSnapshots(leafIndices, leaves, proof, expectedRoot, hash)
+	return valid, err
+}
+
+func ValidatePartialTreeWithParkingSnapshots(leafIndices []uint64, leaves, proof [][]byte, expectedRoot []byte,
```

Comment:
You must add detailed documentation of this function (btw same goes to all exported function)